pypi_formula_mappings.json

```diff
@@ -701,6 +701,10 @@
   "sgr": {
     "exclude_packages": ["certifi", "cryptography"]
   },
+  "sherlock": {
+    "exclude_packages": ["certifi", "numpy"],
+    "extra_packages": ["setuptools"]
```

Comment:
where do you see this being imported? I'm trying to make sure we fix all these issues upstream

Reply:
It's due to `charset-normalizer`.

Reply:
hmm, I'm not seeing any undeclared use of setuptools (i.e. outside of `setup.py`, `pytz` handles it being missing):

```console
$ python3.12 -m venv .venv
$ .venv/bin/pip install sherlock_project
$ rg '^\s*(from|import) (distutils|pkg_resources|setuptools)' .venv/
.venv/lib/python3.12/site-packages/pytz/__init__.py
102:                from pkg_resources import resource_stream

.venv/lib/python3.12/site-packages/numpy/core/tests/examples/limited_api/setup.py
6:from setuptools import setup, Extension

.venv/lib/python3.12/site-packages/numpy/core/tests/examples/cython/setup.py
7:from distutils.core import setup
9:from setuptools.extension import Extension
```

We should be able to remove this and the `start_with` (actually, `start_with: "setuptools"` should never be needed because pip should always install it if not found: https://github.com/pypa/pip/blob/d2a48b2d9ab8d24182c4ba54c00f2a19293211bc/src/pip/_internal/pyproject.py#L104)

assuming you saw an error locally, can you past the stacktrace so I can see look into what happened? I have noticed leftover `pyc` files confuse pip. We could update `std_pip_args` to include `--use-pep517` if that what's happening

Reply:
> ImportError: cannot import name 'setup' from 'setuptools'

That means you have a broken `setuptools` installed (which I'm assuming is from leftover `pyc` files), so `pip` is trying but failing to use it. You can see that with: 

```console
$ python3.12 -c 'import setuptools' # this will succeed
$ python3.12 -c 'from setuptools import setup' # this will fail
$ ls $(brew --prefix)/lib/python3.12/site-packages/setuptools/
__pycache__
```

See https://github.com/Homebrew/brew/issues/13701, `brew cleanup` should remove those files

Reply:
Thanks, that fixed it. I had uninstalled `python-setuptools`, but did not run `brew cleanup` afterward. Thanks for educating me. I'll open a syntax PR to remove that workaround.